src/mapi.ts

```diff
@@ -256,17 +275,46 @@ export class Mapi {
     }
 
     /**
+     * Takes a normalized url and appends a given query string
+     */
+    appendQueryString(url: NormalizedURL, query: string) {
+        url.original += "\\?" + query;
+        url.noTrailing += "?" + query;
+        url.trailing += "?" + query;
+        return url;
+    }
+
+    /**
      * Handles the requests and sends response back accordingly.
      */
     server(ServerRequest: http.ServerRequest, ServerResponse: http.ServerResponse): http.ServerResponse | void {
+        // Add trailing slash no matter what
+        // replace /mapi with /api so that you can define your endpoints as /api but still
+        // use them as / By this way you can have real api and mock api at the same time
+        let reqUrl = this.normalizeUrl(ServerRequest.url);
+
+        if (ServerRequest.method === "POST") {
+            let post = new formidable.IncomingForm();
+
+            post.parse(ServerRequest, function(err, params, files) {
+                var query = queryString.stringify(params);
+                if (query.length > 0) {
+                    //Assumes the POST url is a regex, so we escape the ?
+                    reqUrl = this.appendQueryString(reqUrl, query);
+                }
+                return this.serverHelper(reqUrl, ServerRequest, ServerResponse);
+            }.bind(this));
+        } else {
+            return this.serverHelper(reqUrl, ServerRequest, ServerResponse);
+        }
+    }
+
+
+    serverHelper(reqUrl: NormalizedURL, ServerRequest: http.ServerRequest, ServerResponse: http.ServerResponse): http.ServerResponse | void {
```

Comment:
Instead of renaming this `serverHelper` keep this as `server` just create `getEndpoint` method that receives `ServerRequest` and uses `searchMap` to find endpoint and returns `MapSearchResult` it's basically what you have in `server` method at the moment.